os/se/ameba/security_ameba_wrapper_tz.c

```diff
@@ -0,0 +1,763 @@
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+* either express or implied. See the License for the specific
+* language governing permissions and limitations under the License.
+*
+***************************************************************************************************/
+
+#include <tinyara/config.h>
+#include <semaphore.h>
+#include <tinyara/seclink_drv.h>
+#include <tinyara/security_hal.h>
+#include <tinyara/kmalloc.h>
+
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define AWRAP_TAG "[AMEBA_WRAPPER]"
+#define AWRAP_ENTER                                                                 \
+	do {                                                                             \
+		sedbg(AWRAP_TAG"[INFO] %s %s:%d\n", __FUNCTION__, __FILE__, __LINE__);       \
+	} while (0)
+
+#define HAL_COPY_DATA(in, out, len)                     \
+	do {                                                \
+		if (in->data == NULL) {                         \
+			return HAL_INVALID_ARGS;                    \
+		}                                               \
+		memcpy(in->data, out, len);                     \
+		in->data_len = len;                             \
+	} while (0)
+
+#define HAL_COPY_PRIV_DATA(in, out, len)                \
+	do {                                                \
+		if (in->priv == NULL) {                         \
+			return HAL_INVALID_ARGS;                    \
+		}                                               \
+		memcpy(in->priv, out, len);                     \
+		in->priv_len = len;                             \
+	} while (0)
+
+/* Max Key or Secure Storage Index */
+#define KEY_STORAGE_INDEX_MAX 32
+#define HAL_MAX_RANDOM_SIZE 256
+
+/* Factory Key and Cert define */
+#define FACTORYKEY_RTL_DEVICE (0x00010120)
+#define FACTORYKEY_RTL_CERT (0x00010122)
+
+/* Factory Cert Key */
+#define FACTORY_CERT_ADDR CONFIG_SE_FACTORY_CERT_ADDRESS
+#define FACTORY_KEY_ADDR CONFIG_SE_FACTORY_KEY_ADDRESS
+
+/* Secure Storage Variable */
+#define SEC_STORE_OFFSET CONFIG_SE_SSTORAGE_ADDRESS
+
+/* Secure efuse key location */
+#define SAMSUNG_KEY_ADDR 0x150
+
+/* Non-Secure Data buff */
+#define NS_BUF_LEN 4096
+
+typedef struct {
+	void *ns_func_s;
+	uint8_t *buf;
+	uint32_t buf_len;
+} ns_passin_struc;
+
+typedef struct {
+	hal_data *input_data;
+	hal_data *output_data;
+} inout_struc;
+
+typedef struct {
+	uint32_t factorykey_device_idx;
+	uint32_t factorykey_cert_idx;
+	uint32_t factory_cert_addr;
+	uint32_t factory_key_addr;
+	uint32_t sstorage_addr;
+	uint32_t samsung_key_addr;
+} factory_struc;
+
+uint8_t *ns_buf;
```

Comment:
If this is used in this file only, let's add `static`.

Reply:
Modified.